### PR TITLE
feat: use cosmiconfig, update deps

### LINKED ISSUE #119: Allow the file to be named graphqlconfig.json
Howdy,
I'm working on tools in the GraphQL ecosystem and like the idea to have a unified configuration file that can be shared by many tools.

Unfortunately, I find the editing experience of the .graphqlconfig file to be a bit unpleasant at the moment. 

Issue #52 tracks the idea to provide a JSON schema to allow editors to autocomplete code, which would help a bit. In addition, I want to make the case for allowing the config file to be named `.graphqlconfig.json`:
* It would immediately improve the editing experience in most code editors, checking for syntax errors, highlighting code and providing auto-formatting.
* It is the obvious companion to the already allowed `.graphqlconfig.yaml`
* It would bring home the message that the file format is, in fact, JSON.
* Existing `.graphqlconfig` files would of course still be supported.

It is not clear to me if there is any reason why using the json extension for the config file would not be allowed, but I was also unable to find any previous discussion on the issue.

### PR DESCRIPTION
**Pull request by jedmao**
Uses smart defaults based on conventional expectations in the JS ecosystem

## Deprecated
1. Use `.graphqlrc` or [other cosmiconfig variations](https://www.npmjs.com/package/cosmiconfig) instead of `.graphqlconfig[.y[a]ml]`.

## Breaking Changes
1. `getGraphQLConfig` is now async.
1. `findGraphQLConfigFile` has been removed.

Related: https://github.com/prisma/vscode-graphql/issues/54

fix #119
fix #116
fix #115
fix #110
fix #109
fix #108
fix #107
fix #99
fix #98
fix #95
fix #93
fix #92
fix #89

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,7 +112,7 @@ an endpoint for subscription, you can use expanded version:
 }
 ```
 
-> Note: do not save secure information in .graphqlconfig file. Use [Environment variables](specification.md#referencing-environment-variables) for that like in the example above.
+> Note: do not save secure information in your .graphql file. Use [Environment variables](specification.md#referencing-environment-variables) for that like in the example above.
```
Comment (divyenduz) [inline README.md]: Is this usage correct?

Comment (jedmao) [reply README.md]: Good catch! I need to work on my find/replace skills 😄

**Comment by jedmao** [comment]
This is going to fail until https://github.com/prisma/graphql-import/pull/350 is merged/published.

**Comment by rychkog** [comment]
Any progress on this? Cause the extension is unusable without this fix :(

**Pull request closed without merging**